### DISCUSSION TITLE
Switch node

### DISCUSSION THREAD

**Discussion by MStcz** [zachowj/hass-node-red#168] [Q&A] [answered]
Hi, I am using switch node and I would like to monitor state changes of the node. It is possible to use "Output on state change" which is working fine but just in case the node state is not changed by msg.enable true/false. In that case there is no output message with the state change information. Is it by design or bug? 

Help is saying:
`msg.enable (boolean), Set to true to turn on the switch and false to turn it off. If the message has a property enable defined and set to a boolean the node will not have any output.`
I would expect exception in case I am changing state by msg.enable message.

**Comment by zachowj** [comment]
I don't understand what you're trying to do.

**Reply by MStcz** [reply]
When you use the code below and click on the inject node (toggle) it is working fine, debug messages are displayed in the debug window. But if you click on the inject node (msg.enable - true/msg.enable - false) nothing is displayed even though the switch node state has changed. The question is whether this is by design or bug?


`[{"id":"46559bca6072f114","type":"debug","z":"7cce0378a776386e","name":"debug 20","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":840,"y":200,"wires":[]},{"id":"4f8bb96a25750371","type":"inject","z":"7cce0378a776386e","name":"msg.enable - true","props":[{"p":"enable","v":"true","vt":"bool"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","x":340,"y":200,"wires":[["fef71a680c41d68a"]]},{"id":"63b088ccfd214de1","type":"inject","z":"7cce0378a776386e","name":"msg.enable - false","props":[{"p":"enable","v":"false","vt":"bool"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","x":350,"y":240,"wires":[["fef71a680c41d68a"]]},{"id":"c087685fbd63680c","type":"api-call-service","z":"7cce0378a776386e","name":"","server":"e034745e.c914b8","version":5,"debugenabled":false,"domain":"switch","service":"toggle","areaId":[],"deviceId":[],"entityId":["switch.testswitch"],"data":"","dataType":"jsonata","mergeContext":"","mustacheAltTags":false,"outputProperties":[],"queue":"none","x":590,"y":120,"wires":[[]]},{"id":"2f9b32d8c076029c","type":"inject","z":"7cce0378a776386e","name":"toggle","props":[],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","x":310,"y":120,"wires":[["c087685fbd63680c"]]},{"id":"fef71a680c41d68a","type":"ha-switch","z":"7cce0378a776386e","name":"","version":0,"debugenabled":false,"inputs":1,"outputs":2,"entityConfig":"e1f95e05f83a517a","enableInput":true,"outputOnStateChange":true,"outputProperties":[{"property":"outputType","propertyType":"msg","value":"state change","valueType":"str"},{"property":"payload","propertyType":"msg","value":"","valueType":"entityState"}],"x":570,"y":220,"wires":[["46559bca6072f114"],["b009dbb2de6d4761"]]},{"id":"b009dbb2de6d4761","type":"debug","z":"7cce0378a776386e","name":"debug 21","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":840,"y":240,"wires":[]},{"id":"e034745e.c914b8","type":"server","name":"Home Assistant","version":5,"addon":true,"rejectUnauthorizedCerts":true,"ha_boolean":"y|yes|true|on|home|open","connectionDelay":true,"cacheJson":true,"heartbeat":false,"heartbeatInterval":30,"areaSelector":"friendlyName","deviceSelector":"friendlyName","entitySelector":"friendlyName","statusSeparator":"at: ","statusYear":"hidden","statusMonth":"short","statusDay":"numeric","statusHourCycle":"h23","statusTimeFormat":"h:m","enableGlobalContextStore":true},{"id":"e1f95e05f83a517a","type":"ha-entity-config","server":"e034745e.c914b8","deviceConfig":"","name":"testswitch","version":"6","entityType":"switch","haConfig":[{"property":"name","value":"testswitch"},{"property":"icon","value":""},{"property":"entity_category","value":""},{"property":"device_class","value":""}],"resend":false,"debugEnabled":false}]`

**Comment by zachowj** [comment, marked as answer]
> Help is saying:
> `msg.enable (boolean), Set to true to turn on the switch and false to turn it off. If the message has a property enable defined and set to a boolean the node will not have any output.`

The help section is correct. If you send an input to the switch with the property `enabled` set to a boolean the node will not have any output.
